lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -1076,14 +1076,22 @@ static void optimizeSliceOfSplat(Function *F) {
 }
 
 /// Eliminate ReshapeNode when the input is already the correct shape.
+/// Also eliminate ReshapeNode when the input is SplatNode.
 static void optimizeReshape(Function *F) {
   for (auto &node : F->getNodes()) {
     auto *reshapeNode = dyn_cast<ReshapeNode>(&node);
     if (!reshapeNode)
       continue;
-    auto inputNode = reshapeNode->getInput();
+    auto &inputNode = reshapeNode->getNthInput(0);
     if (inputNode.dims() == reshapeNode->dims()) {
       reshapeNode->getResult().replaceAllUsesOfWith(inputNode);
+      continue;
+    }
+    auto *splatNode = dyn_cast<SplatNode>(inputNode);
+    if (splatNode && splatNode->hasOneUse()) {
+      auto *newSplatNode = F->createSplat(
+          splatNode->getName(), reshapeNode->getType(), splatNode->getValue());
+      reshapeNode->getResult().replaceAllUsesOfWith(newSplatNode);
```

Comment:
nit: I would prefer to add a `continue;` here to keep with the pattern of peephole loops that check different conditions. Would prevent a potential bug if someone adds a new if case below this one and does not realize there is no continue here.

Reply:
Done.